plugins/modules/ndo_macsec_policy.py

```diff
@@ -0,0 +1,443 @@
+                    )
+                    match.details["macsecParams"]["confOffSet"] = "offset{0}".format(confidentiality_offset)
+
+                if key_server_priority and match.details.get("macsecParams")["keyServerPrio"] != key_server_priority:
+                    ops.append(dict(op="replace", path="{0}/{1}/macsecParams/keyServerPrio".format(path, match.index), value=key_server_priority))
+                    match.details["macsecParams"]["keyServerPrio"] = key_server_priority
+
+            if macsec_keys:
+                # updating macsec_keys modifies the existing list with the new list
+                macsec_keys_list = []
+                for macsec_key in macsec_keys:
+                    macsec_keys_list.append(
+                        dict(
+                            keyname=macsec_key.get("key_name"),
+                            psk=macsec_key.get("psk"),
+                            start=mso.verify_time_format(macsec_key.get("start_time")) if macsec_key.get("start_time") else None,
+                            end=mso.verify_time_format(macsec_key.get("end_time")) if macsec_key.get("end_time") else None,
+                        )
+                    )
+
+                if macsec_keys_list != match.details.get("macsecKeys", []):
+                    ops.append(dict(op="replace", path="{0}/{1}/macsecKeys".format(path, match.index), value=macsec_keys_list))
+                match.details["macsecKeys"] = macsec_keys
+            elif macsec_keys == []:
+                # remove macsec_keys if the list is empty
+                ops.append(dict(op="remove", path="{0}/{1}/macsecKeys".format(path, match.index)))
+                match.details.pop("macsecKeys", None)
+
+            mso.sanitize(match.details)
+
+        else:
+            macsec_param_map = {}
+
+            payload = {"name": macsec_policy, "templateId": mso_template.template.get("templateId"), "schemaId": mso_template.template.get("schemaId")}
+            payload["type"] = interface_type
+
+            if description:
+                payload["description"] = description
+            if admin_state:
+                payload["adminState"] = admin_state
+            if cipher_suite:
+                macsec_param_map["cipherSuite"] = cipher_suite
+            if window_size:
+                macsec_param_map["windowSize"] = window_size
+            if security_policy:
+                macsec_param_map["securityPol"] = security_policy
+            if sak_expiry_time:
+                macsec_param_map["sakExpiryTime"] = sak_expiry_time
+
+            if interface_type == "access":
+                if confidentiality_offset:
+                    macsec_param_map["confOffSet"] = "offset{0}".format(confidentiality_offset)
+                if key_server_priority:
+                    macsec_param_map["keyServerPrio"] = key_server_priority
+                payload["macsecParams"] = macsec_param_map
+
+            macsec_keys_list = []
+            if macsec_keys:
```

Comment:
only used in condition
```suggestion
            if macsec_keys:
                macsec_keys_list = []
```